From 4915a9e8e404555379aa8807ad8c53c7589f2ae7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Sun, 8 Sep 2019 18:09:23 +0200
Subject: [PATCH 020/152] iconGrid: Delete private child property when removing
 child

Delete a private property we set when the child got added to make sure
the reference is deleted after the child got removed from the grid.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/704
---
 js/ui/iconGrid.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/ui/iconGrid.js b/js/ui/iconGrid.js
index e0a1e8bd8..08d72aff1 100644
--- a/js/ui/iconGrid.js
+++ b/js/ui/iconGrid.js
@@ -251,6 +251,7 @@ var IconGrid = GObject.registerClass({
 
     _childRemoved(grid, child) {
         child.disconnect(child._iconGridKeyFocusInId);
+        delete child._iconGridKeyFocusInId;
     }
 
     vfunc_get_preferred_width(_forHeight) {
-- 
2.23.0

